docs/sql/sql.md

```diff
@@ -0,0 +1,440 @@
+Note that if a `JOIN` is used,
+an index must be defined on the join column of both tables,
+so that it can be evaluated efficiently.
+
+### WHERE
+
+```ebnf
+predicate
+    = expr
+    | predicate AND predicate
+    ;
+
+expr
+    = literal
+    | column
+    | expr op expr
+    ;
+
+op
+    = '='
+    | '<'
+    | '>'
+    | '<' '='
+    | '>' '='
+    ;
+
+literal
+    = INTEGER
+    | STRING
+    | HEX
+    | TRUE
+    | FALSE
+    ;
+```
+
+While the `SELECT` clause determines the table,
+the `WHERE` clause determines the rows in the subscription.
+
+The subscription language supports conjunctions `AND`.
+It does not support disjunctions `OR`.
+It does not support `!=` comparisons.
+It does not support arithmetic expressions.
+Note that disjunctions can be expressed by subscribing to multiple queries.
+
+## Query and DML
+
+### Statements
+
+- [SELECT](#select)
+- [INSERT](#insert)
+- [DELETE](#delete)
+- [UPDATE](#update)
+- [SET](#set)
+- [SHOW](#show)
+
+### SELECT
+
+```ebnf
+SELECT [ DISTINCT ] projection FROM relation [ [ WHERE predicate ] [ ORDER BY order ] [ LIMIT limit ] ]
+```
+
+#### SELECT Clause
+
+```ebnf
+projection
+    = '*'
+    | table '.' '*'
+    | projExpr { ',' projExpr }
+    | aggrExpr { ',' aggrExpr }
+    ;
+
+projExpr
+    = column [ [ AS ] alias ]
+    ;
+
+aggrExpr
+    = COUNT '(' STAR ')' [AS] alias
+    | COUNT '(' DISTINCT column ')' [AS] alias
+    | SUM   '(' column ')' [AS] alias
+    ;
+```
+
+The `SELECT` clause determines the columns that are returned.
+In particular it supports both column and wildcard projections.
+It also supports sum and count aggregation.
+
+If `DISTINCT` is specified, only unique rows are returned.
+However each projected column must be of a type that allows comparison.
+
+Examples:
+```sql
+-- Select the items in my inventory
+SELECT * FROM Inventory;
+
+-- Select the names of the items in my inventory
+SELECT item_name FROM Inventory
+
+-- How many items are in my inventory?
+SELECT COUNT(*) as num_items FROM Inventory
+```
+
+#### FROM Clause
+
+```ebnf
+FROM table [ [AS] alias ] { [INNER] JOIN table [ [AS] alias ] ON predicate }
+```
+
+The query language supports joins among an arbitrary number of tables.
+It does not impose any index restrictions like the subscription language.
+
+Examples:
+```sql
+-- Which orders include currently stocked items?
+SELECT order.*
+FROM Orders order
+JOIN Inventory inv
+ON order.item_id = inv.item_id
+```
+
+#### WHERE Clause
+
+```ebnf
+predicate
+    = expr
+    | predicate AND predicate
+    | predicate OR  predicate
+    ;
+
+expr
+    = literal
+    | column
+    | expr op expr
+    ;
+
+op
+    = '='
+    | '<'
+    | '>'
+    | '<' '='
+    | '>' '='
+    | '!' '='
+    | '<' '>'
+    ;
+
+literal
+    = INTEGER
+    | FLOAT
+    | STRING
+    | HEX
+    | TRUE
+    | FALSE
+    ;
+```
+
+The query language supports both conjunctions and disjunctions.
+It also supports `!=` comparisons.
+It does not support arithmetic expressions.
```

Comment:
```suggestion
 > TBD: Support arithmetic expressions
```

I think we should clarify that certain features will be added later, no that should not be used.